examples/context/theme-detailed-app.js

```diff
@@ -0,0 +1,37 @@
+import ThemeContext from './theme-context';
+import ThemedButton from './button';
+
+class App extends React.Component {
+  state = {
+    theme: {
+      highlight: 'blue',
+      accent: 'purple',
+    },
+  };
+
+  changeHighlightColor = () => {
+    const colors = ['red', 'blue', 'green'];
+    const randomColor =
+      colors[Math.floor(Math.random() * 3)];
+    this.setState({
+      theme: {
+        ...this.state.theme,
+        highlight: randomColor,
```

Comment:
I'm not a big fan of this randomization.

Maybe we could simplify this a bit with a `toggle` function that flips between "light" and "dark" theme or something?

Reply:
I was hoping to show something with a `value` more complex than a string... maybe an Avatar component with a default pic?

Reply:
Ah, I see. An object value or something seems fine. I just think the randomness maybe detracts from the example some.